starter-code/routes/passportRouter.js

```diff
@@ -1,17 +1,46 @@
 const express        = require("express");
 const passportRouter = express.Router();
-// Require user model
+const passport       = require("passport");
+const LocalStrategy  = require("passport-local").Strategy;
+const User           = require("../models/user");
+const bcrypt         = require("bcrypt");
+const ensureLogin    = require("connect-ensure-login");
 
-// Add bcrypt to encrypt passwords
 
-// Add passport 
+passportRouter.get("/private" , ensureLogin.ensureLoggedIn(), (req,res,next) =>{
+  res.render("passport/private", {user:req.user})
+})
 
+passportRouter.get("/signup", (req, res, next) => {
+  res.render("passport/signup")
+})
 
-const ensureLogin = require("connect-ensure-login");
+passportRouter.post("/signup", (req, res, next) => {
+  console.log('works'+ JSON.stringify(req.body))
+  const { username, password } = req.body;
+      bcrypt.hash(password, 10).then(hash => {
+        User.create({
+          username: username,
+          password: hash
+        }).then(user => {
+          console.log(username)
+          res.render("passport/private", { user });
+        });
+      });
+    });
 
 
-passportRouter.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
-  res.render("passport/private", { user: req.user });
+
+passportRouter.get("/login", (req, res, next) => {
+  res.render("passport/login", { message: req.flash("error") });
 });
+passportRouter.post("/login",
+  passport.authenticate("local", {
```

Comment:
Whenever you see redirect, always remember that you're pointing to a route name. So you should have successRedirect: "/private" and failureRedirect: "/login"